controllers/rabbitmqcluster_controller.go

```diff
@@ -101,6 +102,16 @@ func (r *RabbitmqClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		return ctrl.Result{}, r.prepareForDeletion(ctx, rabbitmqCluster)
 	}
 
+	// exit if skip reconcile annotation is set to true
+	if v, ok := rabbitmqCluster.Annotations["rabbitmq.com/skipReconcile"]; ok && v == "true" {
+		logger.Info("Skip reconciling RabbitmqCluster",
+			"namespace", rabbitmqCluster.Namespace,
+			"name", rabbitmqCluster.Name)
+		r.Recorder.Event(rabbitmqCluster, corev1.EventTypeWarning,
+			"SkipReconcile", "annotation 'rabbitmq.com/skipReconcile' is set to true")
+		return ctrl.Result{}, err
```

Comment:
Shouldn't we update the status before skipping reconciliation?
Skipping reconciliation means we don't update the actual cluster to match the desired spec.
However, reporting the current status should still be done although reconciliation is skipped.
Imagine a pod dies. The `AllReplicasReadyCondition` condition should still be updated to be `false`.

Reply:
Also, in the current implementation deletion is allowed even when reconciliation is skipped.
When reconciliation is skipped and the desired state is `Delete`, the operator shouldn't reconcile, i.e. it shouldn't call `prepareForDeletion`.